src/program/reflection.rs

```diff
@@ -249,7 +249,29 @@ pub unsafe fn reflect_uniforms(ctxt: &mut CommandContext, program: Handle)
         });
     }
 
-    uniforms
+    // Flatten arrays
+    let mut uniforms_flattened = HashMap::new();
+    for uniform in uniforms {
+        // If this is a normal non-array element, just move it over
+        if !uniform.0.ends_with("[0]") {
+            uniforms_flattened.insert(uniform.0, uniform.1);
+            continue;
+        }
+
+        // We've got an array, first get the base of the name
+        let name_base = uniform.0.split_at(uniform.0.len()-3).0;
+        let uniform_base = uniform.1;
+
+        // Go over all the elements in the array
+        for i in 0..uniform_base.size.unwrap() {
+            let mut uniform = uniform_base.clone();
```

Comment:
This could be `let uniform = Uniform { size: None, location: blabla, .. uniform_base };`, which is nicer.